### ISSUE TITLE
How to remove floating window title bars

### ISSUE THREAD

**Issue by SingleKey** [andrewauclair/ModernDocking#136]
I need to drag the panel out of the original window and use a window style without a title bar for the panel, but I don't know how to set it up. Please give me a suggestion. Thank you.

**Comment by andrewauclair** [comment]
That could be interesting. It would have to revert to individual titlebars if there is more than 1 dockable in the window.

**Comment by SingleKey** [comment]
> I'm not sure I follow. The titlebar is always displayed. That's how things are displayed unless you use top tabs.

I want to drag the panel out of the window, and it will become a floating window~

**Comment by SingleKey** [comment]
> That could be interesting. It would have to revert to individual titlebars if there is more than 1 dockable in the window.

I want to use jFrame. setUndecorated(true);

**Comment by andrewauclair** [comment]
That should already be allowed by default. The method responsible is here: https://github.com/andrewauclair/ModernDocking/blob/main/docking-api/src/ModernDocking/Dockable.java#L95.

As long as your Dockable returns true for this method then you should be able to drag and drop outside of the frame to float the Dockable.

**Comment by SingleKey** [comment]
> That could be interesting. It would have to revert to individual titlebars if there is more than 1 dockable in the window.

I think allowing customization would be better~

**Comment by andrewauclair** [comment]
An example of floating can be seen here: https://github.com/andrewauclair/ModernDocking/blob/main/img/floating_panel.gif

**Comment by SingleKey** [comment]
> An example of floating can be seen here: https://github.com/andrewauclair/ModernDocking/blob/main/img/floating_panel.gif

This example includes the window title bar, which is duplicated with the panel title bar. I don't need the window title bar, but I couldn't find the relevant settings~

**Comment by andrewauclair** [comment]
Let me see if I understand now. You want the new frame to be undecorated?

**Comment by SingleKey** [comment]
> Let me see if I understand now. You want the new frame to be undecorated?

Yes, because the window title bar is the same as the panel title bar, I would like to use the panel title bar instead of the window title bar, similar to how panels in similar to Visual Studio or IDEA are dragged out of the window.

**Comment by SingleKey** [comment]
> Let me see if I understand now. You want the new frame to be undecorated?

I found the relevant code:
![test](https://github.com/andrewauclair/ModernDocking/assets/67951198/76e9f721-1cb0-47f4-94ee-088a6c4dfa47)

**Comment by andrewauclair** [comment]
Doing that would mean losing the ability to drag the dockable again as that titlebar is the drag source.

**Comment by SingleKey** [comment]
> Doing that would mean losing the ability to drag the dockable again as that titlebar is the drag source.

Perhaps modifying the window style is not very reasonable. Since the window cannot be modified, can you customize the title bar? I have found the following examples:

https://stackoverflow.com/questions/12822037/how-to-create-customize-title-bar-with-close-button-on-jframe
https://copyprogramming.com/howto/how-to-change-the-color-of-the-title-bar-in-javax-swing
https://www.codespeedy.com/how-to-change-the-color-of-title-bar-in-jframe-in-java/

**Comment by SingleKey** [comment]
> Not sure if this is what's being requested here, but it would be nice if windows with only one dockable would use the window titlebar instead of the custom panel header. E.g. putting the `Inheritance` here into the title bar: ![image](https://private-user-images.githubusercontent.com/48808497/287077456-c9c0709a-6a4b-4028-9d8c-e34dd7aa1032.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE0NDQzNzcsIm5iZiI6MTcwMTQ0NDA3NywicGF0aCI6Ii80ODgwODQ5Ny8yODcwNzc0NTYtYzljMDcwOWEtNmE0Yi00MDI4LTlkOGMtZTM0ZGQ3YWExMDMyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjAxVDE1MjExN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTcxNzJhYzc2M2I5NmNlMGQ5ZmFjMWE0YmZkNGE1ZWQxNzYyNTAxYjhiZjc5MTBhYzE0MmRiMjkyZjdkMzFhOTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.NHNQfnNtsqyOS3MK2VyVUBGF8X5rnv0XWALZ5J3DhMI)
> 
> The three-dot menu would have to be hidden, but in this state it contains no useful elements anyway

I think in this situation, the title of the panel can be set to the window, including those three points.
